debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java

```diff
@@ -175,6 +181,32 @@ public List<T> addDataCollectionNamesToSnapshot(List<String> dataCollectionIds)
         return newDataCollectionIds;
     }
 
+    @Override
+    public void stopSnapshot() {
+        stopRequested = true;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void removeDataCollectionsFromSnapshot(List<String> dataCollectionIds) {
+        this.dataCollectionsToBeRemoved.addAll(dataCollectionIds.stream()
+                .map(x -> (T) TableId.parse(x, useCatalogBeforeSchema))
+                .collect(Collectors.toList()));
+    }
+
+    @Override
+    public void applyPendingDataCollectionIdRemovals() {
+        if (!dataCollectionsToBeRemoved.isEmpty()) {
+            Iterator<T> iterator = dataCollectionsToBeRemoved.iterator();
+            while (iterator.hasNext()) {
+                final T dataCollectionId = iterator.next();
+                LOGGER.info("Removed '{}' from incremental snapshot collection list.", dataCollectionId);
+                dataCollectionsToSnapshot.remove(dataCollectionId);
```

Comment:
Here should be a log message that a data colection for which snapshot should be stopped is not present.